ratis-server-api/src/main/java/org/apache/ratis/server/raftlog/RaftLogSequentialOps.java

```diff
@@ -125,10 +125,21 @@ <OUTPUT, THROWABLE extends Throwable> OUTPUT runSequentially(
 
   /**
    * Append asynchronously an entry.
-   * Used by the leader.
+   * Used by the leader for scenarios that there is no needs to clean up resources when the given entry is no
+   * longer used/referenced by this log.
    */
   default CompletableFuture<Long> appendEntry(LogEntryProto entry, TransactionContext context) {
-    return appendEntry(entry);
+    return appendEntry(ReferenceCountedObject.wrap(entry), context);
+  }
```

Comment:
The method is not used anywhere except for the new `append(ReferenceCountedObject)`.  It should throw an exception and annotated as deprecated.
```java
  /**
   * @deprecated use {@link #append(ReferenceCountedObject)}.
   */
  @Deprecated
  default CompletableFuture<Long> appendEntry(LogEntryProto entry, TransactionContext context) {
    throw new UnsupportedOperationException();
  }
```